trin-core/src/portalnet/overlay.rs

```diff
@@ -305,7 +305,9 @@ where
                     XorMetric::distance(
                         &content_key_value.0.content_id(),
                         &node.key.preimage().raw(),
-                    ) < node.value.data_radius()
+                    )
+                    .0[3]
+                        < node.value.data_radius().as_u64()
```

Comment:
I don't understand what bug we are fixing here. IIUC, this changes the comparison to a comparison between `u64`, where we take the most significant 64 bits of the distance as a `u64` and convert the data radius from a `U256` to a `u64`. This seems like it would introduce a bug.

Also, the call to `U256::as_u64` will [panic](https://docs.rs/ethereum-types/0.13.1/ethereum_types/struct.U256.html#method.as_u64) if the data radius value is greater than `u64::MAX`.

We should not merge this as is.

Reply:
I think there are some confusing discrepancies in data radius in our codebase:

Data radius is defined as a `U256` value in the portal network[ specs](https://github.com/ethereum/portal-network-specs/blob/master/history-network.md#radius) but in portal storage, we use u64: https://github.com/ethereum/trin/blob/43c2527c7f941a015b9d7b322f1a139fcf9a53c0/trin-core/src/portalnet/storage.rs#L41.
https://github.com/ethereum/trin/blob/43c2527c7f941a015b9d7b322f1a139fcf9a53c0/trin-core/src/portalnet/storage.rs#L132

Another confusing thing is how we set the default data radius values in `overlay_service`:
https://github.com/ethereum/trin/blob/43c2527c7f941a015b9d7b322f1a139fcf9a53c0/ethportal-peertest/src/jsonrpc.rs#L20

https://github.com/ethereum/trin/blob/43c2527c7f941a015b9d7b322f1a139fcf9a53c0/trin-core/src/portalnet/overlay_service.rs#L394

https://github.com/ethereum/trin/blob/43c2527c7f941a015b9d7b322f1a139fcf9a53c0/trin-core/src/portalnet/overlay_service.rs#L1039

I guess the default data radius value should be `U256::MAX` instead of `U256::from(u64::MAX)` and we need to update the storage value to `U256`?

`XorMetric::distance` returns a `U256` value which most of the time is bigger than u64 which makes the filter above ignore all nodes in the routing table because of their default data radius as `u64::MAX`.

I guess reverting this change and adding a default data radius constant as `U256::MAX` + portal storage fix will solve the issue.

Reply:
This is what #305 was meant to address. I can resurrect that PR, or close it and open a more isolated change. I would like to revisit that and see if it can be improved in any way. I sort of lost track of it after the start of Devconnect.